### ISSUE TITLE
unusedArguments stripping out used ones

### ISSUE THREAD

**Issue by victorBaro** [nicklockwood/SwiftFormat#1694]
When using new swift conditional variable unwrapping definition (i.e. `let myOptional` instead of `let myOptional = myOptional`), swiftFormat treats it as an unused argument.

Take the following code:

```swift
        listenForUpdates() { [weak self] update, error in
            guard let update, error == nil else {
                return
            }

            self?.configure(update)

        }
```

When using `--stripunusedargs closure-only`, it produces the following result, which results on a compiler error:

```swift
        listenForUpdates() { [weak self] _, error in // update variable gets replaced by _
            guard let update, error == nil else {
                return
            }

            self?.configure(update)

        }
```

Note that when using `guard let update = update`, swiftformat respects the variable and it does not replace it with _.

Tested with swiftformat v 0.53.8

**Comment by nicklockwood** [comment]
@victorBaro I suspect this is the same issue as https://github.com/nicklockwood/SwiftFormat/issues/1688, in which case it's already fixed, but I'll verify that before the next release

**Comment by nicklockwood** [comment]
@victorBaro fixed in 0.53.9
